### PR TITLE
Implemented Policies page

### PR DESCRIPTION
**Pull request by SteffiAntony**
Jira: https://jsw.ibm.com/browse/PFEBMC-2395
Signed-off-by: Steffi Antony <“steffiantony196@gmail.com”>

### CODE REVIEW

File: src/views/SecurityAndAccess/Policies/Policies.vue [hunk 1]
```diff
@@ -1,465 +1,441 @@
...
               </span>
-              <span v-if="hostUsbState">
+              <span v-if="Policies.hostUsbEnabled">
                 {{ $t('global.status.enabled') }}
               </span>
               <span v-else>{{ $t('global.status.disabled') }}</span>
-            </b-form-checkbox>
-          </b-col>
-        </b-row>
-        <b-row v-if="isAdminUser || isServiceUser" class="section-divider">
```
Comment (nabilpandiyalayil) [inline src/views/SecurityAndAccess/Policies/Policies.vue:219]: This condition is required

File: src/views/SecurityAndAccess/Policies/Policies.vue [hunk 2]
```diff
@@ -1,465 +1,441 @@
...
+        </BRow>
+        <BRow class="section-divider">
+          <BCol class="d-flex align-items-center justify-content-between">
             <dl class="mt-3 mr-3 w-75">
               <dt>
                 {{ $t('pagePolicies.hostUsb') }}
-                <info-tooltip :title="$t('global.status.nextReboot')">
-                  <icon-time />
-                </info-tooltip>
+                <span v-b-tooltip="$t('global.status.nextReboot')">
```
Comment (nabilpandiyalayil) [inline src/views/SecurityAndAccess/Policies/Policies.vue]: Try to use InfoTooltip global component here

File: src/views/SecurityAndAccess/Policies/Policies.vue [hunk 3]
```diff
@@ -1,465 +1,441 @@
...
               <dt>{{ $t('pagePolicies.usbFirmwareUpdatePolicy') }}</dt>
               <dd>
                 {{ $t('pagePolicies.usbFirmwareUpdatePolicyDescription') }}
               </dd>
             </dl>
-            <b-form-checkbox
+            <BFormCheckbox
               id="usbFirmwareUpdatePolicySwitch"
-              v-model="usbFirmwareUpdatePolicyState"
-              :disabled="!(isAdminUser || isServiceUser)"
```
Comment (nabilpandiyalayil) [inline src/views/SecurityAndAccess/Policies/Policies.vue:149]: This condition is required.

File: src/views/SecurityAndAccess/Policies/Policies.vue [hunk 4]
```diff
@@ -1,465 +1,441 @@
 <template>
-  <b-container fluid="xl">
+  <div class="container" fluid="xl">
```
Comment (nabilpandiyalayil) [inline src/views/SecurityAndAccess/Policies/Policies.vue]: Use BContainer

**Review by nabilpandiyalayil — CHANGES_REQUESTED** [review]

File: src/views/SecurityAndAccess/Policies/Policies.vue [hunk 5]
```diff
@@ -1,465 +1,448 @@
...
+          </BCol>
+        </BRow>
+        <BRow class="section-divider">
+          <BCol class="d-flex align-items-center justify-content-between">
             <dl class="mt-3 mr-3 w-75">
               <dt>
                 {{ $t('pagePolicies.vtpm') }}
                 <info-tooltip :title="$t('global.status.nextReboot')">
-                  <icon-time />
+                  <status-icon status="time" />
```
Comment (Nikhil-Ashoka) [inline src/views/SecurityAndAccess/Policies/Policies.vue]: Icon is bigger. Make it 16*16px. It is currently 20*20px

File: src/views/SecurityAndAccess/Policies/Policies.vue [hunk 6]
```diff
@@ -1,465 +1,448 @@
...
-    </b-row>
-  </b-container>
+            </BFormCheckbox>
+          </BCol>
+        </BRow>
+      </BCol>
+    </BRow>
+    <BModal
+      v-model="modal"
+      title="Unauthenticated ACF upload enablement"
```
Comment (Nikhil-Ashoka) [inline src/views/SecurityAndAccess/Policies/Policies.vue]: Use from translation file

File: src/views/SecurityAndAccess/Policies/Policies.vue [hunk 7]
```diff
@@ -1,465 +1,448 @@
...
+      cancel-title="cancel"
+      ok-title="confirm"
```
Comment (Nikhil-Ashoka) [inline src/views/SecurityAndAccess/Policies/Policies.vue]: "cancel" and "confirm" take from the translation file

File: src/views/SecurityAndAccess/Policies/Policies.vue [hunk 8]
```diff
@@ -1,465 +1,448 @@
...
+      @cancel="onModalCancel"
+      @ok="onModalOk"
```
Comment (Nikhil-Ashoka) [inline src/views/SecurityAndAccess/Policies/Policies.vue:257]: Size of cancel and confirm buttons are different.

File: src/views/SecurityAndAccess/Policies/Policies.vue [hunk 9]
```diff
@@ -1,465 +1,448 @@
...
-        </b-row>
-      </b-col>
-    </b-row>
-  </b-container>
+            </BFormCheckbox>
+          </BCol>
+        </BRow>
+      </BCol>
+    </BRow>
+    <BModal
```
Comment (Nikhil-Ashoka) [inline src/views/SecurityAndAccess/Policies/Policies.vue:250]: Try to implement in composables here. will be required for upcoming implementations.

File: src/views/SecurityAndAccess/Policies/Policies.vue (hunk 5 above)
Comment (nabilpandiyalayil) [inline src/views/SecurityAndAccess/Policies/Policies.vue]: I think including time as a status icon is a bit confusing. Can we use the icon-time component in here like it was before? StatusIcon component should only have  icons representing status.

File: src/components/Global/StatusIcon.vue [hunk 10]
```diff
@@ -4,16 +4,19 @@
...
 </template>
 
 <script setup>
+import { defineProps } from 'vue';
 import IconInfo from '@carbon/icons-vue/es/information--filled/20';
 import IconSuccess from '@carbon/icons-vue/es/checkmark--filled/20';
 import IconWarning from '@carbon/icons-vue/es/warning--filled/20';
-import IconSecondary from '@carbon/icons-vue/es/error--filled/20';
 import IconDanger from '@carbon/icons-vue/es/misuse/20';
+import IconTime from '@carbon/icons-vue/es/time/20';
```
Comment (nabilpandiyalayil) [inline src/components/Global/StatusIcon.vue]: I'm not comfortable with this change. This is not technically a status icon.

Comment (nabilpandiyalayil) [inline src/components/Global/StatusIcon.vue]: This is no longer required. Please remove.

File: src/components/Global/StatusIcon.vue [hunk 11]
```diff
@@ -43,6 +46,9 @@ defineProps({
   &.warning {
     color: $warning;
   }
+  &.time {
+    color: #2d60e5; // theme-color('time');
+  }
```
Comment (nabilpandiyalayil) [inline src/components/Global/StatusIcon.vue]: This one also.

**Review by nabilpandiyalayil — CHANGES_REQUESTED** [review]

File: src/views/SecurityAndAccess/Policies/Policies.vue [hunk 12]
```diff
@@ -199,267 +200,279 @@
...
+      ref="myModalRef"
+      v-model="modal"
+      :title="$t('pagePolicies.acfUploadEnablement')"
+      cancel-title="cancel"
+      ok-title="confirm"
```
Comment (nabilpandiyalayil) [inline src/views/SecurityAndAccess/Policies/Policies.vue]: Use from the translation file.

**Review by nabilpandiyalayil — CHANGES_REQUESTED** [review]

**Review by nabilpandiyalayil — APPROVED** [review]

**Review by Nikhil-Ashoka — APPROVED** [review]

**Pull request merged**